pages/api/courses/[courseId]/layouts/index.ts

```diff
@@ -0,0 +1,51 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "../../../../../lib/prisma";
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { courseId: courseIdParam } = req.query;
```

Comment:
Wonder if we could abstract this logic out into the lib folder since it is going to be nearly identical across so many routes?

Reply:
Added, and to save a PR I added it to existing spots as well